clippy_lints/src/misc.rs

```diff
@@ -457,7 +456,7 @@ fn check_nan(cx: &LateContext<'_, '_>, expr: &Expr<'_>, cmp_expr: &Expr<'_>) {
                     cx,
                     CMP_NAN,
                     cmp_expr.span,
-                    "doomed comparison with `NAN`, use `std::{f32,f64}::is_nan()` instead",
```

Comment:
This must have been an error a long time? The `is_nan` method is not available in the library modules, it's only available on the primitive types.